armi/cli/reportsEntryPoint.py

```diff
@@ -0,0 +1,170 @@
+        if app is None:
+            raise RuntimeError("NEED APP!")
+        pm = app._pm
+
+        if self.args.h5db is None:
+            # Just do begining stuff, no database is given...
+            if self.cs is not None:
+                cs = self.cs
+                settings.setMasterCs(self.cs)
+                blueprint = blueprints.loadFromCs(cs)
+                r = factory(cs, blueprint)
+                report.title = r.name
+            else:
+                raise RuntimeError(
+                    "No Settings with Blueprint or Database, cannot gerenate a report"
+                )
+
+            with directoryChangers.ForcedCreationDirectoryChanger("reportsOutputFiles"):
+
+                pluginContent = pm.hook.getReportContents(
+                    r=r,
+                    cs=cs,
+                    report=report,
+                    stage=ReportStage.Begin,
+                    blueprint=blueprint,
+                )
+                site = report.writeReports()
+                if self.args.view:
+                    webbrowser.open(site)
+
+        else:
+            db = databaseFactory(self.args.h5db, "r")
+            if self.args.bp is not None:
+                blueprint = self.args.bp
+            i = 0
+
+            with db:
+
+                with directoryChangers.ForcedCreationDirectoryChanger(
+                    "reportsOutputFiles"
+                ):
+
+                    dbNodes = list(db.genTimeSteps())
+                    cs = db.loadCS()
+                    if self.args.bp is None:
+                        blueprint = db.loadBlueprints()
+                    r = factory(cs, blueprint)
+                    report.title = r.name
+                    pluginContent = (
+                        armi.getPluginManagerOrFail().hook.getReportContents(
+                            r=r,
+                            cs=cs,
+                            report=report,
+                            stage=ReportStage.Begin,
+                            blueprint=blueprint,
+                        )
+                    )
+                    for cycle, node in dbNodes:
+                        if nodes is not None and (cycle, node) not in nodes:
+                            continue
+
+                        if (
+                            self.args.min_node is not None
+                            and (cycle, node) < self.args.min_node
+                        ):
+                            continue
+
+                        if (
+                            self.args.max_node is not None
+                            and (cycle, node) > self.args.max_node
+                        ):
+                            continue
+
+                        r = db.load(cycle, node)
+                        cs = db.loadCS()
+
+                        if i == len(dbNodes) - 1:
+                            stage = ReportStage.End
+                        else:
+                            stage = ReportStage.Standard
```

Comment:
Is this right? Seems like you still need standard processing for the last node, but extra end processing, too. So always do Standard, then outside the loop do another call to the plugin hood, passing `End`